solutions/USACO jan19-grass-plantin.mdx

```diff
@@ -0,0 +1,102 @@
+---
+id: usaco-894
+source: USACO Silver 2019 January
+title: Grass Planting
+author: Aadit Ambadkar
+---
+
+## Solution 1 - Intended Solution
+
+$\mathcal{O}(N)$ using key insight.
+
+Let ```adj[i]``` represent the number of paths adjacent to node $i$ in the graph. Then, considering that node, and that node alone, you would need exactly ```adj[i]+1``` numbers to fill it. This is because you would need ```adj[i]``` for each of the adjacent nodes, and $1$ for the node itself. So, our answer is the maximum of all the adjacents, plus $1$. 
```

Comment:
i think the offical editorial explains it well enough

Reply:
I was requested by other people to add an editorial to my solutions. Could you please confer with them first before requesting this. Or should the rules be changed?

Reply:
oh. Who may I ask? 👀